posthog/hogql_queries/insights/trends/test/__snapshots__/test_trends.ambr

```diff
@@ -251,22 +251,36 @@
                   groups.group_key AS key
            FROM groups
            WHERE and(equals(groups.team_id, 2), ifNull(equals(index, 0), 0))
            GROUP BY groups.group_type_index,
                     groups.group_key) AS e__group_0 ON equals(e.`$group_0`, e__group_0.key)
         WHERE and(equals(e.team_id, 2), equals(e.event, 'sign up'), greaterOrEquals(toTimeZone(e.timestamp, 'UTC'), toDateTime64('2020-01-02 00:00:00.000000', 6, 'UTC')), less(toTimeZone(e.timestamp, 'UTC'), toDateTime64('2020-01-03 00:00:00.000000', 6, 'UTC')), ifNull(equals(e__group_0.properties___industry, 'technology'), 0)))
-     GROUP BY actor_id) AS source
+     GROUP BY actor_id SETTINGS use_query_cache=1,
+                                query_cache_ttl=600) AS source
   INNER JOIN
-    (SELECT argMax(person.created_at, person.version) AS created_at,
+    (SELECT argMax(toTimeZone(person.created_at, 'UTC'), person.version) AS created_at,
             person.id AS id
      FROM person
-     WHERE equals(person.team_id, 2)
+     WHERE and(equals(person.team_id, 2), in(id,
+                                               (SELECT source.actor_id AS actor_id
+                                                FROM
+                                                  (SELECT actor_id AS actor_id, count() AS event_count, groupUniqArray(100)(tuple(timestamp, uuid, `$session_id`, `$window_id`)) AS matching_events
+                                                   FROM
+                                                     (SELECT e.person_id AS actor_id, toTimeZone(e.timestamp, 'UTC') AS timestamp, e.uuid AS uuid, e.`$session_id` AS `$session_id`, e.`$window_id` AS `$window_id`
+                                                      FROM events AS e
+                                                      LEFT JOIN
+                                                        (SELECT argMax(replaceRegexpAll(nullIf(nullIf(JSONExtractRaw(groups.group_properties, 'industry'), ''), 'null'), '^"|"$', ''), toTimeZone(groups._timestamp, 'UTC')) AS properties___industry, groups.group_type_index AS index, groups.group_key AS key
+                                                         FROM groups
+                                                         WHERE and(equals(groups.team_id, 2), ifNull(equals(index, 0), 0))
+                                                         GROUP BY groups.group_type_index, groups.group_key) AS e__group_0 ON equals(e.`$group_0`, e__group_0.key)
+                                                      WHERE and(equals(e.team_id, 2), equals(e.event, 'sign up'), greaterOrEquals(toTimeZone(e.timestamp, 'UTC'), toDateTime64('2020-01-02 00:00:00.000000', 6, 'UTC')), less(toTimeZone(e.timestamp, 'UTC'), toDateTime64('2020-01-03 00:00:00.000000', 6, 'UTC')), ifNull(equals(e__group_0.properties___industry, 'technology'), 0)))
+                                                   GROUP BY actor_id SETTINGS use_query_cache=1, query_cache_ttl=600) AS source)))
```

Comment:
This case of the optimization kicking in seems massive, but seems legit at a glance. I do wonder how fast this is, given the `in()`'s SELECT is triple-nested. 😅 Probably depends on whether the events time range is large (can be months for an aggregate query) or tiny (a single day).

Reply:
https://github.com/PostHog/posthog/pull/23135#issuecomment-2182830571

This sort of got buried but discussed the performance difference with Marius above. I think this will (almost) always make things faster, because if you have enough events to have the inner queries run slow, you will also have a lot of people for the inefficient join.

If this ends up slowing down small users, we can figure out a way to create a set of "large user optimizations" that maybe only get turned on once people hit some event limit. 

This will be *much* faster once materialized CTEs get landed in Clickhouse.